virttest/utils_windows/drive.py

```diff
@@ -37,3 +38,75 @@ def get_floppy_drives_letter(session):
     """
     cond = "MediaType!=0 AND MediaType!=11 AND MediaType!=12"
     return _logical_disks(session, cond=cond, props=["DeviceID"])
+
+
+def rescan_disks(session):
+    """
+    Rescan disks in windows guest.
+
+    :param session: Session object.
+    """
+    script_path = r"%TEMP%\rescan.dp"
+    rescan_cmd = "echo rescan > {0} && diskpart /s {0}"
+    session.cmd(rescan_cmd.format(script_path))
+
+
+def get_drive_volume_num(session, disk_letter):
+    """
+    Get a drive volume num in windows guest.
+
+    :param session: Session object.
+    :param disk_letter: Destination volume location.
+
+    :return: drive volume's number.
+    """
+    script_path = r"%TEMP%\gvolume.dp"
+    list_disk_cmd = "echo list disk > {0} && diskpart /s {0}"
+    disks = session.cmd_output(list_disk_cmd.format(script_path))
+    index_list = re.findall(r"DISK (\d)", disks, re.I)
+
+    detail_cmd = 'echo select disk {0} > {1} && echo detail disk >> {1} '
+    detail_cmd += '&& diskpart /s {1}'
+    gvol_pattern = r"Volume (\d)\s+%s" % disk_letter
+
+    for index in index_list:
+        details = session.cmd_output(detail_cmd.format(index, script_path))
+        volume_num = re.findall(gvol_pattern, details, re.M)
+        if volume_num:
+            return volume_num[0]
+
+
+def extend_volume(session, volume_num):
+    """
+    Extend a volume in windows guest.
+    Default extend to max usage size.
+
+    :param session: Session object.
+    :param volume_num: drive volume's number.
+    """
+    script_path = r"%TEMP%\extend.dp"
+    extend_cmd = 'echo select volume %s > {0} && ' % volume_num
+    extend_cmd += 'echo extend >> {0} && diskpart /s {0}'
+    session.cmd(extend_cmd.format(script_path))
+
+
+def shrink_volume(session, volume_num, size):
+    """
+    Shrink a volume in windows guest.
+    Default shrink to max usage size.
+
+    :param session: Session object.
+    :param volume_num: drive volume's number.
+    :param size: Desired decrease size.
+    """
+    script_path = r"%TEMP%\shrink.dp"
+    select_cmd = 'echo select volume %s > {0} && ' % volume_num
+    query_cmd = 'echo shrink querymax >> {0} && diskpart /s {0}'
+    query_cmd = ' '.join([select_cmd, query_cmd])
+    query_output = session.cmd_output(query_cmd.format(script_path))
+    max_num = re.findall("(\d+) MB", query_output)[0]
+    if int(max_num) >= int(size):
```

Comment:
Seems not need to verify the value, I mean, if the given `size` is invalid, let the command run and report error directly.